src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerDifferenceService.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.diff
+
+import org.apache.commons.lang3.builder.Diff
+import org.apache.commons.lang3.builder.DiffBuilder
+import org.apache.commons.lang3.builder.DiffResult
+import org.apache.commons.lang3.builder.ToStringStyle
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import kotlin.reflect.full.findAnnotations
+
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class DiffableProperty(val type: PropertyType)
+
+enum class PropertyType {
+  IDENTIFIERS, PERSONAL_DETAILS, STATUS, LOCATION, SENTENCE, RESTRICTED_PATIENT
+}
```

Comment:
Categorise Prisoner properties into different types. Eventually we will publish events telling consumers which property types have changed so they can update if they are interested.